packages/store/addon/-private/system/ds-model-store.ts

```diff
@@ -0,0 +1,383 @@
+    }
+    assert(`You need to pass a model name to the store's modelFor method`, isPresent(modelName));
+    assert(
+      `Passing classes to store methods has been removed. Please pass a dasherized string instead of ${modelName}`,
+      typeof modelName === 'string'
+    );
+
+    let maybeFactory = this._modelFactoryFor(modelName);
+
+    // for factorFor factory/class split
+    let klass = maybeFactory.class ? maybeFactory.class : maybeFactory;
+    if (!klass.isModel) {
+      return new ShimModelClass(this, modelName);
+    } else {
+      return klass;
+    }
+  }
+
+  _modelFactoryFor(modelName) {
+    if (DEBUG) {
+      assertDestroyedStoreOnly(this, '_modelFactoryFor');
+    }
+    assert(`You need to pass a model name to the store's _modelFactoryFor method`, isPresent(modelName));
+    assert(
+      `Passing classes to store methods has been removed. Please pass a dasherized string instead of ${modelName}`,
+      typeof modelName === 'string'
+    );
+    let normalizedModelName = normalizeModelName(modelName);
+    let factory = getModelFactory(this, this._modelFactoryCache, normalizedModelName);
+
+    if (factory === null) {
+      throw new EmberError(`No model was found for '${normalizedModelName}'`);
+    }
+
+    return factory;
+  }
+
+  /*
+  Returns whether a ModelClass exists for a given modelName
+  This exists for legacy support for the RESTSerializer,
+  which due to how it must guess whether a key is a model
+  must query for whether a match exists.
+
+  We should investigate an RFC to make this public or removing
+  this requirement.
+
+  @private
+ */
+  _hasModelFor(modelName) {
+    if (DEBUG) {
+      assertDestroyingStore(this, '_hasModelFor');
+    }
+    assert(`You need to pass a model name to the store's hasModelFor method`, isPresent(modelName));
+    assert(
+      `Passing classes to store methods has been removed. Please pass a dasherized string instead of ${modelName}`,
+      typeof modelName === 'string'
+    );
+
+    if (CUSTOM_MODEL_CLASS) {
+      return this.getSchemaDefinitionService().doesTypeExist(modelName);
+    } else {
+      assert(`You need to pass a model name to the store's hasModelFor method`, isPresent(modelName));
+      assert(
+        `Passing classes to store methods has been removed. Please pass a dasherized string instead of ${modelName}`,
+        typeof modelName === 'string'
+      );
+      let normalizedModelName = normalizeModelName(modelName);
+      let factory = getModelFactory(this, this._modelFactoryCache, normalizedModelName);
+
+      return factory !== null;
+    }
+  }
+
+  _relationshipMetaFor(modelName, id, key) {
+    if (CUSTOM_MODEL_CLASS) {
+      return this._relationshipsDefinitionFor(modelName)[key];
+    } else {
+      let modelClass = this.modelFor(modelName);
+      let relationshipsByName = get(modelClass, 'relationshipsByName');
+      return relationshipsByName.get(key);
+    }
+  }
+
+  _attributesDefinitionFor(modelName) {
```

Comment:
double-check the `identifiers` codepath for retrieving attributes does not pull from this info

Reply:
(same for relationships definition)